### ISSUE TITLE
[Mesheryctl] Installation script fails

### ISSUE THREAD

**Issue by kumarabd** [layer5io/meshery#2680]
#### Description
Mesheryctl installation script fails with the following error:

```
$ curl -L https://git.io/meshery | bash -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  3156  100  3156    0     0   4401      0 --:--:-- --:--:-- --:--:--  4401
Downloading mesheryctl-v0.5.1 for Linux...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     9  100     9    0     0     49      0 --:--:-- --:--:-- --:--:--    49
Extracting mesheryctl-v0.5.1...
Archive:  /root/meshery.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /root/meshery.zip or
        /root/meshery.zip.zip, and cannot find /root/meshery.zip.ZIP, period.
Installing mesheryctl in /usr/local/bin.
mv: cannot stat '/root/mesheryctl': No such file or directory
mv: cannot stat '/root/mesheryctl': No such file or directory
Cannot install mesheryctl. Check permissions of root for /usr/local/bin.
```

#### Environment:
- Meshery Version: v0.5.1
- Kubernetes System Type: docker-desktop
- Host OS:
- Browser:

**Comment by hexxdump** [comment]
I tried this locally on a Linux m/c.

```
Extracting mesheryctl-v0.5.1...
+ unzip -o /root/mesh/test/meshery.zip
Archive:  /root/mesh/test/meshery.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /root/mesh/test/meshery.zip or
        /root/mesh/test/meshery.zip.zip, and cannot find /root/mesh/test/meshery.zip.ZIP, period.
+ install_dir=
+ '[' -z '' ']'
+ install_dir=/usr/local/bin
+ printf '\nInstalling mesheryctl in /usr/local/bin.\n'

Installing mesheryctl in /usr/local/bin.
++ whoami
+ WHOAMI=root
+ mv /root/mesh/test/mesheryctl /usr/local/bin/mesheryctl
mv: cannot stat '/root/mesh/test/mesheryctl': No such file or directory
+ sudo mv /root/mesh/test/mesheryctl /usr/local/bin/mesheryctl
```

unzip command is failing on Linux OS. 
I do not think its due to the recently introduced "-o" option to unzip, as the unzip command fails even without -o option.

```
# unzip meshery.zip
Archive:  meshery.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of meshery.zip or
        meshery.zip.zip, and cannot find meshery.zip.ZIP, period.
```

Checked whether the meshery.zip file is correct or not.

```
+ URL=https://github.com/meshery/meshery/releases/download/v0.5.1/mesheryctl_0.5.1_Linux_x86_64.zip
+ printf '\nDownloading %s for %s...\n\n' mesheryctl-v0.5.1 Linux

Downloading mesheryctl-v0.5.1 for Linux...

+ curl -L https://github.com/meshery/meshery/releases/download/v0.5.1/mesheryctl_0.5.1_Linux_x86_64.zip -o /root/mesh/test/meshery.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     9  100     9    0     0     18      0 --:--:-- --:--:-- --:--:--    18
+ printf '\nExtracting %s...\n' mesheryctl-v0.5.1
```

```
# file meshery.zip
meshery.zip: ASCII text, with no line terminators
# cat meshery.zip
Not Found
```
The issue is with https://github.com/meshery/meshery/releases/download/v0.5.1/ path. It throws 404.

**Comment by kumarabd** [comment]
@hexxdump Are you pushing the fix for this tonight?

**Comment by hexxdump** [comment]
@kumarabd 
I see that the new github path for release v0.5.1 has only the source.zip file - https://github.com/meshery/meshery/archive/refs/tags/v0.5.1.zip
But not the platform specific zip files. This could be an issue with how the https://github.com/meshery/meshery/releases/ page is updated.

```
NAME="mesheryctl-${MESHERY_VERSION}"
URL="https://github.com/meshery/meshery/releases/download/${MESHERY_VERSION}/mesheryctl_${MESHERY_VERSION:1}_${OSEXT}_${OSARCHITECTURE}.zip"
```

The fix should be to copy the OS and architecture specific zip files under https://github.com/meshery/meshery/releases/download/v0.5.1/ path correctly by GoReleaser, right?

**Comment by hwpplayer1** [comment]
On Ubuntu 20.04.2 Focal

```
hwpplayer1@masscollabs:~$ curl -L https://git.io/meshery | bash -
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
100  3156  100  3156    0     0   1605      0  0:00:01  0:00:01 --:--:--  4745

Downloading mesheryctl-v0.5.1 for Linux...

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100     9  100     9    0     0     11      0 --:--:-- --:--:-- --:--:--    11

Extracting mesheryctl-v0.5.1...
Archive:  /home/hwpplayer1/meshery.zip
  End-of-central-directory signature not found.  Either this file is not
  a zipfile, or it constitutes one disk of a multi-part archive.  In the
  latter case the central directory and zipfile comment will be found on
  the last disk(s) of this archive.
unzip:  cannot find zipfile directory in one of /home/hwpplayer1/meshery.zip or
        /home/hwpplayer1/meshery.zip.zip, and cannot find /home/hwpplayer1/meshery.zip.ZIP, period.

Installing mesheryctl in /usr/local/bin.
mv: cannot stat '/home/hwpplayer1/mesheryctl': No such file or directory
[sudo] password for hwpplayer1: 
mv: cannot stat '/home/hwpplayer1/mesheryctl': No such file or directory
Cannot install mesheryctl. Check permissions of hwpplayer1 for /usr/local/bin.
hwpplayer1@masscollabs:~$ 
```

**Comment by leecalcote** [comment]
@hwpplayer1 test again on v0.5.2, if you would.

**Comment by leecalcote** [comment]
@Gorim @aneeshnema, the issue of installing `mesheryctl` is fixed in today's v0.5.2 release. Please find 

```
curl -L https://git.io/meshery | bash -
``` 

back in working order. 👍
